crates/types/src/partition_table.rs

```diff
@@ -282,6 +379,22 @@ impl PartitionTableBuilder {
         }
     }
 
+    pub fn for_each<F>(&mut self, mut modify: F)
+    where
+        F: FnMut(&PartitionId, &mut TargetPartitionReplicationState),
+    {
+        for (partition_id, partition) in self.inner.partitions.iter_mut() {
+            let mut replica_group_builder = partition.replica_group.clone().into();
+            modify(partition_id, &mut replica_group_builder);
+            let replica_group = ReplicaGroup::from(replica_group_builder);
+
+            if partition.replica_group != replica_group {
```

Comment:
This is a bug I am working on fixing it. It seems the index set will equal even if the items are the same order. So change of leader only will appear as `not modified`